msteams-platform/toolkit/use-CICD-template.md

```diff
@@ -10,233 +10,357 @@ ms.date: 04/20/2022
+* [Set up CI/CD pipelines using your own workflow](#set-up-cicd-pipelines-using-your-own-workflow). *[Optional]*
 
-## Set up workflows with GitHub
+## Set up CI/CD pipelines with Teams Toolkit CLI
 
-To set up pipelines with GitHub for CI/CD:
+> [!NOTE]
+> Use Teams Toolkit version 5.6.0 or a later version.
+
+You can use [Teams Toolkit command line interface (CLI)](Teams-Toolkit-CLI.md) to set up CI/CD pipeline for your Teams app.
+
+### Prerequisites
+
+| **Item** | **Description** |
+| --- | --- |
+| Set up required resources for your Teams app, such as Teams app ID, bot ID, and so on. | • Manually extract the resources from the `manifest.json` file under the `appPackage` folder. <br> • Automatically generate to run the `Provision` command in Teams Toolkit. |
+| Configure Azure resources |• Manually prepare the resources by examining the bicep files under the `infra` folder. <br> • Automatically prepare the resources using the `Provision` command in Teams Toolkit.|
+| Ensure that you've a service principal and its access policies on resources are properly configured. Set up a service principal as follows:| • [Create service principal using Entra portal](/entra/identity-platform/howto-create-service-principal-portal). <br> • [Create service principal using Azure CLI](/cli/azure/azure-cli-sp-tutorial-1?tabs=bash). <br> • The `Teamsapp` command-line interface (CLI) supports Azure login with a service principal secret. [Create a secret](/entra/identity-platform/howto-create-service-principal-portal) and save the client ID, client secret, and tenant ID of the service principal. <br> :::image type="content" source="../assets/images/teams-toolkit-v2/service-principal.png" alt-text="Screenshot shows the service principal secret.":::|
+
+After you've completed the prerequisites, let's set up a pipeline:
+
+* [Set up pipeline with GitHub](#set-up-pipeline-with-github).
+
+* [Set up pipeline with Azure DevOps](#set-up-pipeline-with-azure-devops).
 
-* Create CI/CD workflows.
-* Customize CI/CD workflows.
+### Set up pipeline with GitHub
 
-### Create CI/CD workflows
+To set up the pipeline with GitHub, follow these steps:
 
-1. Download the corresponding template files from [Tools and Templates](#tools-and-templates).
-1. Rename the downloaded template files by your needs.
-1. Put them under `.github/workflows`, which is the designated folder for GitHub Actions.
-1. Commit and push these template files into remote repositories.
-1. Add necessary [encrypted secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) for your workflows.
-1. Trigger your workflows. Check more [details](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow) about how to trigger a workflow on GitHub.
+1. Open Visual Studio Code.
```

Comment:
```suggestion
```
It doesn't need to be VScode. Any IDE will suffice.

Reply:
Since the article refers to Visual Studio Code (VSC) and explains the steps involved, it would be incorrect to remove the ‘Open Visual Studio Code’ step from the procedure.